[ci] MI355 runner hanging during tests

From [CI workflow](https://github.com/ROCm/TheRock/actions/runs/16305988414/job/46087953062).

For GitHub runners, we notice these errors:

```
FAILED tests/test_rocm_sanity.py::TestROCmSanity::test_hip_printf - Exception: b'Memory access fault by GPU node-2 (Agent handle: 0x562d2b5c1610) on address (nil). Reason: Write access to a read-only page.\n'
```

For ssh testing, the executable hangs at [`hipLaunchKernelGGL`](https://github.com/ROCm/TheRock/blob/142fcbc288251e34585535eab941191a1837dd73/tests/hipcc_check.cpp#L14) after hipcc compiles that file


To reproduce for hipcc test:

```
$ git clone https://github.com/ROCm/TheRock.git
$ cd TheRock
$ python3 build_tools/install_rocm_from_artifacts.py --run-id 16350224221 --amdgpu-family gfx950-dcgpu --tests
$ export THEROCK_BIN_DIR=./therock-build/bin/
$ ./${THEROCK_BIN_DIR}/hipcc tests/hipcc_check.cpp -Xlinker -rpath=${THEROCK_BIN_DIR}../lib/ -o test_hipcc
$ ./test_hipcc
```

To reproduce for ctest or gtest:
```
$ git clone https://github.com/ROCm/TheRock.git
$ cd TheRock
$ python3 build_tools/install_rocm_from_artifacts.py --run-id 16350224221 --amdgpu-family gfx950-dcgpu --tests
$ sudo apt install cmake
 // for ctest
$ cd therock-build/bin/rocprim
$ ctest
// for gtest
cd therock-build/bin
$ ./rccl-UnitTests
```

artifacts pulled from [this commit](https://github.com/ROCm/TheRock/commit/8189f38d44874f2dacd35fe505685e3b39ee34a7)

MI355 were working on [this commit](https://github.com/ROCm/TheRock/commit/f686688e59531b5ea59c67ffd49b1e2ac7a15d54), but failed on [this commit](https://github.com/ROCm/TheRock/commit/daafcd0b82a1cb2484c6c3c907d179377900c3eb)

--- a/build_tools/github_actions/amdgpu_family_matrix.py
+++ b/build_tools/github_actions/amdgpu_family_matrix.py
@@ -24,7 +24,7 @@ amdgpu_family_info_matrix_presubmit = {
 amdgpu_family_info_matrix_postsubmit = {
     "gfx950": {
         "linux": {
-            "test-runs-on": "",  # Disabling due to issues (https://github.com/ROCm/TheRock/issues/1049), label: linux-mi355-1gpu-ossci-rocm
+            "test-runs-on": "linux-mi355-1gpu-ossci-rocm",
             "family": "gfx950-dcgpu",
         }
     },

